tests/difference/core/README.md

```diff
@@ -1,135 +0,0 @@
-# Differential testing for Interchain Security 'core' protocol
-
```

Comment:
This README is obsolote very soon (https://github.com/cosmos/interchain-security/blob/df7ea65610458eaf964bbaf75f7a00b8003f89bf/tests/mbt/README.md), so I just deleted it instead of going through and fixing the links.